File: Ywy/JavaScript/level2/test/StringCompression.test.js
import StringCompression from '../StringCompression';

test(`"abcabcabcabcdededededede" is 14`, () => {
    expect(new StringCompression().solution("abcabcabcabcdededededede")).toBe(14);
});

test(`"xababcdcdababcdcd" is 17`, () => {
    expect(new StringCompression().solution("xababcdcdababcdcd")).toBe(17);
});

test(`"ababcdcdababcdcd" is 9`, () => {
    expect(new StringCompression().solution("ababcdcdababcdcd")).toBe(9);
});

test(`"aabbaccc" is 7`, () => {
    expect(new StringCompression().solution("aabbaccc")).toBe(7);
});

File: Ywy/JAVA/src/level2/test/cases/StringCompressionTest.java
package level2.test.cases;

import static org.junit.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;

import level2.StringCompression;

public class StringCompressionTest {
	private String[] inputs;
	private int[] answers;
	private StringCompression stringCompression;
	@Before
	public void setUp() throws Exception {
		inputs = new String[] {
			"aabbaccc",
			"ababcdcdababcdcd",
			"abcabcdede",
			"abcabcabcabcdededededede",
			"xababcdcdababcdcd"
		};
		answers = new int[] { 7, 9, 8, 14, 17 };
		
		stringCompression = new StringCompression();
	}

	@Test
	public void test() {
		int index = 0;
		int range = inputs.length;
		int result = -1;
		int answer = 0;
		do {
			result = stringCompression.solution(inputs[index]);
			answer = answers[index];
			assertEquals("success: "+ answer + ", result: " + result, result, answer);
		} while((++index) < range);
	}

}
